src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs

```diff
@@ -4871,6 +4875,11 @@ void EmitLoop(RegexNode node)
                         // startingPos = base.runstack[--runstack];
                         EmitStackPop();
                         Stloc(iterationCount);
+                        if (startingStackpos is not null)
+                        {
+                            EmitStackPop();
+                            Stloc(startingStackpos);
```

Comment:
Small NIT: Should we update the comment above to also add the popping of startigStackpos?

Reply:
Yup, thanks.